app/packages/core/src/components/Filters/categoricalFilter/filterOption/FilterOption.tsx

```diff
@@ -208,32 +253,95 @@ const FilterOption: React.FC<Props> = ({
   const onSelectItem = (key: Key) => {
     setKey(key);
     setOpen(false);
+    switch (key) {
+      case "filter":
+        onSelectFilter();
+        break;
+      case "negativeFilter":
+        onSelectNegativeFilter();
+        break;
+      case "match":
+        onSelectMatch();
+        break;
+      case "negativeMatch":
+        onSelectNegativeMatch();
+        break;
+      case "visible":
+        onSelectVisible();
+        break;
+      case "notVisible":
+        onSelectNotVisible();
+        break;
+    }
   };
 
   const onSelectFilter = () => {
-    setExcluded && setExcluded(false);
-    setIsMatching && setIsMatching(false);
-    setOnlyMatch && setOnlyMatch(true);
+    excluded && setExcluded(false);
+    isMatching && setIsMatching(false);
+    !onlyMatch && setOnlyMatch(true);
   };
 
   const onSelectNegativeFilter = () => {
-    setExcluded && setExcluded(true);
-    setIsMatching && setIsMatching(false);
-    setOnlyMatch && setOnlyMatch(false);
+    !excluded && setExcluded(true);
+    isMatching && setIsMatching(false);
+    onlyMatch && setOnlyMatch(false);
   };
 
   const onSelectMatch = () => {
-    setExcluded && setExcluded(false);
-    setIsMatching && setIsMatching(true);
-    setOnlyMatch && setOnlyMatch(true);
+    excluded && setExcluded(false);
+    !isMatching && setIsMatching(true);
+    !onlyMatch && setOnlyMatch(true);
   };
 
   const onSelectNegativeMatch = () => {
-    setExcluded && setExcluded(true);
-    setIsMatching && setIsMatching(true);
-    setOnlyMatch && setOnlyMatch(true);
+    !excluded && setExcluded(true);
+    !isMatching && setIsMatching(true);
+    !onlyMatch && setOnlyMatch(true);
   };
 
+  const onSelectVisible = () => {
+    excluded && setExcluded(false);
+    onlyMatch && setOnlyMatch(false);
+  };
+
+  const onSelectNotVisible = () => {
+    !excluded && setExcluded(true);
+    onlyMatch && setOnlyMatch(false);
+  };
+
+  useEffect(() => {
```

Comment:
Can this initialization effect be avoided? Ideally this state has a default automatically, and is updated through explicit user interaction.

Reply:
This hook always runs on on initial render and changes state. It was part of the reason for the bug fixed by https://github.com/voxel51/fiftyone/pull/3249

Reply:
Ah, I see what you mean. I can fix it.